### ISSUE TITLE
Is there a preHook Function for onNodeSelect() ?

### ISSUE THREAD

**Issue by manojp1988** [primefaces/primeng#5186]
**I'm submitting a feature report**  (check one with "x")
```
[] bug report => Search github for a similar issue or PR before submitting
[X] feature request => Please check if request is not on the roadmap already https://github.com/primefaces/primeng/wiki/Roadmap
[ ] support request => Please do not submit support request here, instead see http://forum.primefaces.org/viewforum.php?f=35
```

**Plunkr Case (Bug Reports)**
http://plnkr.co/edit/6EamO1muJywad7grgUwO?p=preview

**Current behavior**
OnNodeClick() will just select the node. 

**Expected behavior**
OnNodeClick() should let the parent decide whether to allow or not. Some kind of hook is required.

**Minimal reproduction of the problem with instructions**
see **plunkr** 

**What is the motivation / use case for changing the behavior?**
I am using routing on node select. When I move away and select different node, `canDeactivate `trigger will call based on my form. If it returns false, the new node selection should not occur. It should stay selected in the old node. Currently, new node stayed selected.

Basically I don't want to move away if returns false, and `onNodeSelect() `should not get called.

**Please tell us about your environment:**
 Windows 10, VSCODE, npm , dev server comes with ng serve, ...

* **Angular version:** 5.2.5

* **PrimeNG version:** 5.2.0

* **Browser:** [all ]

 
* **Language:** [TypeScript 2.4.2]

* **Node (for AoT issues):** `node --version` =8.9.4

**Comment by manojp1988** [comment]
I misunderstood the  onNodeUnSelect trigger. I guess it will be triggered in multiselection mode. 

But for my scenario, I have added a prehook function before onNodeClick().. if this preHook function is true, I allow onNodeClick to execute, else won't.

Not sure, if this is the right approach. But it should work for my usecase.

Hence closing this ticket.
